File: discovery-service/src/main/java/com/sudo248/discoveryservice/repository/entity/SupplierProduct.java
package com.sudo248.discoveryservice.repository.entity;

import lombok.*;

import javax.persistence.*;

@Data
@Entity
@Table(name = "supplierProduct")
@NoArgsConstructor
public class SupplierProduct {
    @EmbeddedId
    SupplierProductEmbeddable embeddedId;

    @ManyToOne
    @MapsId("productId")
    @JoinColumn(name="product_id")
    private Product product;

    @ManyToOne
    @MapsId("supplierId")
    @JoinColumn(name="supplier_id")
    private Supplier supplier;

    @Column(name="amount_left")
    private int amountLeft;

    @Column(name="price")
    private double price;

    @Column(name = "sold_amount")
    private double soldAmount;

    @Column(name = "rate")
    private double rate;
}

File: client/Soc/app/src/main/java/com/sudo248/soc/domain/entity/discovery/SupplierProduct.kt
package com.sudo248.soc.domain.entity.discovery

data class SupplierProduct(
    val supplierId: String,
    val productId: String,
    val route: Route,
    val amountLeft: Int,
    val price: Double,
    val soldAmount: Int,
    val rate: Double,
)